src/cogent3/core/alignment.py

```diff
@@ -554,6 +554,23 @@ def copy(self):
         result = self.__class__(self, moltype=self.moltype, info=self.info)
         return result
 
+    def deepcopy(self, sliced=True, force_same_data=True):
+        """Returns deep copy of self."""
+        new_seqs = dict()
+        for seq in self.seqs:
+            try:
+                new_seq = seq.deepcopy(sliced=sliced)
+            except:
+                new_seq = seq.copy()
+            new_seqs[seq.name] = new_seq
+            seq_names = list(new_seqs.keys())
+
+        if force_same_data:
```

Comment:
this is violating the immutability principle of this object -- self should never modify it's underlying data